src/coreclr/vm/jitinterface.cpp

```diff
@@ -3969,54 +3969,13 @@ CorInfoType CEEInfo::getTypeForPrimitiveValueClass(
     }
     else
     {
-        switch (th.GetInternalCorElementType())
+        CorElementType elementType = th.GetInternalCorElementType();
+        if (elementType >= ELEMENT_TYPE_BOOLEAN && elementType <= ELEMENT_TYPE_R8
```

Comment:
We have a helper method for this. I think it should be possible to simplify the whole thing into:
```
```
TypeHandle th(clsHnd);
_ASSERTE (!th.IsGenericVariable());

CorElementType elementType = th.GetVerifierCorElementType();
if (CorIsPrimitiveType(elementtype) || elementType == ELEMENT_TYPE_PTR || ELEMENT_TYPE_FNPTR)
{
    result = asCorInfoType(elementType);
}
```

It looks odd that this method is handling `ELEMENT_TYPE_PTR` and `ELEMENT_TYPE_FNPTR`. It does not match the name `getTypeForPrimitiveValueClass`. 

Is the JIT really expecting this method to handle `ELEMENT_TYPE_PTR` and `ELEMENT_TYPE_FNPTR`?

Reply:
> Is the JIT really expecting this method to handle `ELEMENT_TYPE_PTR` and `ELEMENT_TYPE_FNPTR`?
Yes, for example in https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/importer.cpp#L4203

Reply:
`CorIsPrimitiveType` would also include `ELEMENT_TYPE_END`, would that be okay?

Reply:
> Yes, for example in https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/importer.cpp#L4203

I do not see the connection between this line and the method being changed in this PR. Could you please shed some light on it?

> CorIsPrimitiveType would also include ELEMENT_TYPE_END, would that be okay?

Yes, that' ok. `ELEMENT_TYPE_END` is never returned by as CorElementType of a type.

Reply:
`CORINFO_FLG_VALUECLASS` is not set for pointers and function pointers. This is unreachable code that can be deleted.

Any other places?

Reply:
https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/importer.cpp#L3596 relies on it not being undef
https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/importer.cpp#L13647 same here
https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/importer.cpp#L15730 and here

I also see some usages in interpreter.cpp.

Reply:
None of these places will be affected if `elementType == ELEMENT_TYPE_PTR || elementType == ELEMENT_TYPE_FNPTR` check is deleted here. Could you please change the condition to just `if (CorIsPrimitiveType(elementType))` so that it matches the method name, and see whether it is going to cause any issues?

Reply:
The failure looks like the same as https://github.com/dotnet/runtime/pull/66413#issuecomment-1175489442 which might be related to https://github.com/dotnet/runtime/pull/71439. Does not look related to your change.

Reply:
The failure is unrelated and failed in other PRs - https://github.com/dotnet/runtime/pull/66413#issuecomment-1175509710

Reply:
I have opened blocking-clean-in issue https://github.com/dotnet/runtime/issues/71684 on this.

When merging on red, failures should be always tracked by active blocking-clean-in issue.